<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html>

<head>

    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-133572971-1"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'UA-133572971-1');
    </script>

    <!-- Google Analytics -->
    <!--script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

        ga('create', 'UA-44649121-2', 'auto');
        ga('send', 'pageview');
    </script-->
    <!-- End Google Analytics -->

    <title>Zhi-Luo Deng CV</title>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
    <meta name="keywords" content="CV, Zhi-Luo Deng, Data science, Bioinformatician, Microbiome">
    <meta name="description" content="Zhi-Luo Deng's CV">
    <link rel="stylesheet" type="text/css" href="style/reset-fonts-grids.css" media="all">
    <link rel="stylesheet" type="text/css" href="style/resume.css" media="all">
    <style type="text/css">
        .info {
            font-size: 17px;
            font-weight: bold;
            width: 60%;
            float: left;
            line-height: 23px
        }

        .photo {
            font-size: 17px;
            font-weight: bold;
            /*width: 20%;*/
            float: right;
            line-height: 23px
        }

        img {
            margin: 0px;
        }

        h1 {
            font-size: 25px;
        }

        div.yui-u.first {
            width: 100%;
        }

        .job {
            padding-bottom: 5px;
        }

        .job h3 {
            font-size: 16px;
            padding-top: 5px;
            border: none;
        }

    </style>
</head>

<body>
    <div id="doc2" class="yui-t7">
        <div id="inner">
            <div id="hd">
                <div class="yui-gc">
                    <div class="yui-u first">

                        <h1>Zhi-Luo Deng</h1>
                        <div class="info">
                            <br> Gender: Male
                            <br>Birthday: Nov. 23rd, 1987
                            <br>Nationality: Chinese
                            <br>E-mail: <a href="mailto:dawnmsg@gmail.com">dawnmsg(at)gmail.com</a>
                            <!-- <br>Online CV: <a href="https://dawnmy.github.io/CV_new/">https://dawnmy.github.io/CV_new/</a> -->
                            <br>Address: Braunschweig, Germany
                            <!-- <br>Research interests: Complex diseases, Microbiome, Evolutionary biology, Bioinformatics -->
                        </div>
                        <div class="photo">
                            <img src="./style/avator.jpg" height=150px/>
                        </div>
                    </div>
                </div>
            </div>

            <div id="bd">
                <div id="yui-main">
                    <div class="yui-b">
                        <div class="yui-gf">
                            <div class="yui-u first">
                                <h2>Education</h2>
                            </div>
                            <div class="yui-u">
                                <div class="enlarge">
                                    <h3 class="stitle">Postdoctoral researcher (2018~2021)</h3>
                                    <div class="degree">
                                        Research area: Bioinformatics, Human microbiome
                                        <br>Helmholtz Centre for Infection research, Germany
                                        <br>Group: Prof. Dr. Alice McHardy (Computational biology)
                                    </div>
                                    <h3 class="stitle">Doctor of Science (2014~2018)</h3>
                                    <div class="degree">
                                        Research area: Human microbiome, Periodontitis, IBD
                                        <br>Helmholtz Centre for Infection research, Germany
                                        <br>Supervisor: Prof. Dr. Irene Wagner-Döbler
                                        <br>Thesis: Activity profiling of human microbiota in health and disease using metatranscriptomics. (Excellent defense) 
                                    </div>
                                    <h3 class="stitle">Master of Science (2010~2013)</h3>
                                    <div class="degree">Major: Biochemistry and Molecular Biology
                                        <br>Research area: Bioinformatics, Cheminformatics
                                        <br>College of Life Sciences &amp; Technology, Huazhong Agricultural University
                                        <br>Supervisor: Prof. Hong-Yu Zhang &amp; Prof. De-Xin Kong
                                    </div>
                                    <h3 class="stitle">Bachelor of Engineering (2006~2010)</h3>
                                    <div class="degree">Major: Biological Engineering
                                        <br>College of Life Sciences &amp; Technology, Huazhong Agricultural University
                                    </div>
                                </div>
                            </div>
                        </div>
                        <!--// .yui-gf -->

                        <div class="yui-gf">
                            <div class="yui-u first">
                                <h2>Research Skills</h2>
                            </div>
                            <div class="yui-u">

                                <div class="talent">
                                    <h2>Programing and computational techniques</h2>
                                    <p>
                                        Proficient in Linux shell, Python, R. Using Pytorch, Snakemake, Docker. Experienced in Web development and machine learning including deep learning.
                                    </p>
                                    

                                </div>
                                <div class="talent">
                                    <h2>Bioinformatics and research skills </h2>
                                        <p>(meta-)transcriptomics, (meta-)genomics, proteomics, Hi-C, network analysis, phylogenetics, cheminformatics and computer aidded drug discovery.
                                    </p>
                                </div>

                                <h3 style="margin-top:25px;"></h3>
                            </div>

                        </div>
                        <!--// .yui-gf -->

                        <div class="yui-gf" style="border-top: 1px solid #CCCCCC;border-bottom: 1px solid #CCCCCC;
  padding-top: 1em;">
                            <div class="yui-u first">
                                <h2>Projects</h2>
                            </div>

                            <div class="yui-u" style="text-align:justify;">
                            <div class="job">
                                <h2>
                                    1. QuasiModo - benchmarking of strain-resolved analysis for population of large DNA viruses
                                    </h2>
                                    <p>
                                    We developed a benchmarking tool QuasiModo and extensively evaluated 15 assemblers and five six variant callers on ten lab-generated benchmark data sets created with two different library preparation protocols, to identify best practices and challenges for analyzing such data.
                                    (Contribution: developed the software, benchmarked all methods in the study) </p>

                                </div>
                                <div class="job">
                                <h2>
                                    2. Deep neural networks optimized for genomic datasets
                                    </h2>
                                    <p>
                                    We developed a software based on deep learning (Tensorflow, Pytorch) to learn genomic patterns from sequence data. It can be adapted into different use cases, including sequence classification, gene prediction, genomic structure identification, et. al. (Contribution: implemented data visualization, built classification model)
                                    </p>
                                </div>

                                <div class="job">
                                    <h2>3. Development of software for accurate pathogen detection and taxonomy profiling on long reads metagenomics data</h2>
                                    <p>
                                    We are developing computational method using Python, Snakemake and C++ for high resolution pathogen and taxonomy profiling. The software will predict antibiotic resistance for the identified pathogens for precision medicine.
                                    </p>
                                </div>
                               
                                <div class="job">
                                <h2>
                                    4. Second round of challenges for Critical Assessment of Metagenome Interpretation (CAMI)
                                    </h2>
                                    <p>
                                    We established a set of benchmarking criteria, tool-kits and generated datasets for the assessment of metagenomic data anaylsis methods used for genome assembly, binning and taxonomy profiling. (Contribution: evaluated pathogen detection and profiling methods) 
                                    </p>
                                </div>
                                <div class="job">
                                    <h2>5. Explore biologically relevant chemical space for drug discovery (2011 ~ 2013)</h2>
                                    <p>
                                    We proposed a novel multi-dimensional molecular descriptor termed Biological Relevance Spectrum (BRS). Several prediction models (ADME/T and biological activity) built based on BRS using SVM reach both high Precision and Recall. (Contribution: data collection, built classification models)
                                    </p>
                                </div>

                                <div class="job last">
                                <h2>
                                    6. AutoChrom3D: a webserver for modeling and visualization the 3D structure of human or mouse chromatin
                                    </h2>
                                    <p>We built a webserver to predict the 3D structure of chromatin based on Hi-C, 3C, 4C data. (Contribution: built the webserver) </p>
                                </div>

                                <br>
                            </div>
                        </div>

                        <div class="yui-gf last" style="padding-top:6px;">
                            <div class="yui-u first">
                                <h2>Selected publications</h2>
                            </div>
                            <div class="yui-u">
                                <ol id="pub" type="1">
                                     <li>
                                      <strong>Deng ZL</strong>, Dhingra A, Fritz A, Götting J, Münch PC, Steinbrück L, Schulz T, Ganzenmueller T, McHardy AC. Evaluating assembly and variant calling software for strain-resolved analysis of large DNA-viruses.
 <i>Briefings in Bioinformatics</i>. 2020. (In press, IF: 11.6).
                                    </li>
                                    
                                    <li>
                                      <strong>Deng ZL</strong>, Gottschick C, Bhuju S, Masur C, Abels C, Wagner-Doebler I. Metatranscriptome analysis of the vaginal microbiota reveals potential mechanisms for protection against metronidazole in bacterial vaginosis.
                                        <i>mSphere</i>. 2018;3. (<strong>Corresponding author and co-first author</strong>, IF: 4.4)
                                    </li>
                                    <li>
                                      <strong>Deng ZL</strong>, Sztajer H, Jarek M, Bhuju S, Wagner-Doebler I. Worlds apart - transcriptome profiles of key oral microbes in the periodontal pocket compared to single laboratory culture reflect synergistic interactions.
                                        <i>Frontiers in Microbiology</i>. 2018;9. (<strong>Corresponding author</strong>, IF: 5.6)
                                    </li>
                                    <li><strong>Deng ZL</strong>, Szafranski SP, Jarek M, Bhuju S, Wagner-Döbler I. Dysbiosis in chronic periodontitis: Key microbial players and interactions with the human host. <i>Scientific Reports</i>. 2017;7. (IF: 5.1)
                                    </li>
                                    <li><strong>Deng ZL</strong>, Du CX, Li X, Hu B, Kuang ZK., Wang R, Feng SY, Zhang HY and Kong DX. Exploring the biologically relevant chemical space for drug discovery. <i>Journal of chemical information and modeling</i>. 2013;53. (IF: 5.0)
                                    </li>
                                    <li>Gottschick C, <strong>Deng ZL</strong>, Vital M, Masur C, Abels C, Pieper DH, Rohde M, Mendling W, Wagner-Doebler I. Treatment of biofilms in bacterial vaginosis by an amphoteric tenside pessary-clinical study and microbiota analysis. <i>Microbiome</i>. 2017;5. (IF: 13.6)
                                    </li>
                                    <li>Gottschick C, <strong>Deng ZL</strong>, Vital M, Masur C, Abels C, Pieper DH, Wagner-Doebler I. The urinary microbiome of men and women and its changes in women during bacterial vaginosis and antibiotic treatment. <i>Microbiome</i>. 2017;5. (IF: 13.6)
                                    </li>
                                     <li>
                                        Szafranski SP, <strong>Deng ZL</strong>, Tomasch J, Jarek M, Bhuju S, Meisinger C, Kühnisch J, Sztajer H, Wagner-Döbler I. Functional biomarkers for chronic periodontitis and insights into the roles of Prevotella nigrescens and Fusobacterium nucleatum; a metatranscriptome analysis. <i>npj Biofilms and Microbiomes</i>. 2015;1. (IF: 6.8)
                                    </li>
                                    <li>
                                        Szafrański SP, <strong>Deng ZL</strong>, Tomasch J, Jarek M, Bhuju S, Rohde M, Sztajer H, Wagner-Döbler I. Quorum sensing of Streptococcus mutans is activated by Aggregatibacter actinomycetemcomitans and by the periodontal microbiome. <i>BMC genomics</i>. 2017;18. (IF: 4.0)
                                    </li>
                                    <li>
                                        Fritz A, Bremges A, <strong>Deng ZL</strong>, Lesker LR, Ganzenmüller T, Klawonn F, Dilthey A, McHardy A. Haploflow: strain-resolved de novo assembly of viral genomes. <i>Genome Biology</i>. 2021;212 (IF: 13.5) 
                                    </li>

                                    <li>
                                        Wang C, <strong>Deng ZL</strong>, Xie ZM, Chu XY, Chang JW, Kong DX, Li BJ, Zhang HY and Chen LL. Construction of a genome-scale metabolic network of the plant pathogen <i>Pectobacterium carotovorum</i> provides new strategies for bactericide discovery.
                                        <i>FEBS Letters</i>. 2014;893 (<strong>co-first author</strong>, IF: 4.1)
                                    </li>


                                    <li>
                                        Yang XX,
                                        <strong>Deng ZL</strong>, Liu R. RBRDetector: Improved prediction of RNA-binding residues from protein structures using complementary feature- and template-based strategies. <i>Proteins</i>. 2014;82 (IF: 3.8)
                                    </li>
                                   
                                    <li>
                                        Milici M, <strong>Deng ZL</strong>, Tomasch J, Decelle J, Wos-Oxley ML, Wang H, Jáuregui R, Plumeier I, Giebel HA, Badewien TH, Wurst M. Co-occurrence analysis of microbial taxa in the Atlantic Ocean reveals high connectivity in the free-living bacterioplankton. <i>Frontiers in microbiology</i>. 2016;7. (IF: 5.6)
                                    </li>
                                    <li>
                                        Reck M, Tomasch J, <strong>Deng ZL</strong>, Jarek M, Husemann P, Wagner-Döbler I. Stool metatranscriptomics: A technical guideline for mRNA stabilisation and isolation. <i>BMC genomics</i>. 2015;4. (IF: 4.0)
                                    </li>
                                    <li>Peng C, Fu LY, Dong PF,
                                        <strong>Deng ZL</strong>, Li JX, Wang XT and Zhang HY. The sequencing bias relaxed characteristics of Hi-C derived data and implications for chromatin 3D modeling.
                                        <i>Nucleic Acids Research</i>. 2013;41. (IF: 17.0)
                                    </li>

                                    <!--li>
                                        Milici M, Tomasch J, Wos-Oxley ML, Decelle J, Jáuregui R, Wang H, <strong>Deng ZL</strong>, Plumeier I, Giebel HA, Badewien TH, Wurst M. Bacterioplankton Biogeography of the Atlantic Ocean: A Case Study of the Distance-Decay Relationship. <i>Frontiers in microbiology</i>. 2016;7.
                                    </li>
                                    <li>
                                        Milici M, Tomasch J, Wos-Oxley ML, Wang H, Jáuregui R, Camarinha-Silva A, <strong>Deng ZL</strong>, Plumeier I, Giebel HA, Wurst M, Pieper DH. Low diversity of planktonic bacteria in the tropical ocean. <i>Scientific reports</i>. 2016;6.
                                    </li>
                                    <li>
                                        Conrads G, Wendt LK, Hetrodt F, <strong>Deng ZL</strong>, Pieper D, Abdelbary MM, Barg A, Wagner-Döbler I, Apel C. Deep sequencing of biofilm microbiomes on dental composite materials. <i>Journal of oral microbiology</i>. 2019;11.
                                    </li-->
                                    <br/>
                                    <h3>In preparation</h3>
                                    <hr>
                                    

                                    <li>
                                      <strong>Deng ZL</strong>, Reck M, Lange K, Buening J, Sina C, Jarek M, Vital M, Stallmach A, Pieper DH, Wagner-Döbler I. Metatranscriptome analysis of fecal microbiota transplantation in chronic antibiotic resistant pouchitis. <i>Microbiome</i>. (Under review)
                                    </li>
                                    <li>
                                      <strong>Deng ZL</strong>, Münch PC, McHardy AC. Rapid and accurate detection of ribosomal RNA sequences using deep-learning. <i>Nucleic Acids Research</i>. (Under revision)
                                    </li>
                                    <li>
                                    CAMI community. Critical Assessment of Metagenome Interpretation - the second round of benchmarking challenges. <i>Nature methods</i>. (Accepted)
                                    </li>

                                </ol>
                            </div>
                        </div>
                    </div>
                </div>
            </div>

        </div>

    </div>
    <!--<script src="style/analytics.js" async=""></script>
    <script>
        (function(i, s, o, g, r, a, m) {
            i['GoogleAnalyticsObject'] = r;
            i[r] = i[r] || function() {
                (i[r].q = i[r].q || []).push(arguments)
            }, i[r].l = 1 * new Date();
            a = s.createElement(o),
                m = s.getElementsByTagName(o)[0];
            a.async = 1;
            a.src = g;
            m.parentNode.insertBefore(a, m)
        })(window, document, 'script', '//www.google-analytics.com/analytics.js', 'ga');
        ga('create', 'UA-44649121-2', 'dawnmy.github.io');
        ga('send', 'pageview');
    </script>-->
</body>

</html>
